Cython/Compiler/Nodes.py

```diff
@@ -8159,6 +8159,36 @@ def cython_view_utility_code():
     'inspect': ("__Pyx_patch_inspect", "PatchInspect", "Coroutine.c"),
 }
 
+def cimport_numpy_check(node, code):
+    # shared code between CImportStatNode and FromCImportStatNode
+    # check to ensure that import_array is called
+    if node.module_name != "numpy":
+        return
```

Comment:
I think it's better to only call this function if we really cimport from numpy. It should be explicit in the cimport nodes what we special case and what not.
```suggestion
```